Xplat/src/main/java/vazkii/botania/common/helper/ForcePushHelper.java

```diff
@@ -0,0 +1,26 @@
+package vazkii.botania.common.helper;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ForcePushHelper {
+	private static final ThreadLocal<AtomicInteger> forcePushCounter = ThreadLocal.withInitial(AtomicInteger::new);
+
+	public static boolean isForcePush() {
+		return forcePushCounter.get().get() > 0;
+	}
+
+	public static ForcePushContext createForcePushContext() {
+		return new ForcePushContext();
```

Comment:
why not just make the constructor public and call it directly? This is just a needless indirection.